### PR TITLE
add signing intent to koji build metadata

### PR DESCRIPTION
**Pull request by rcerven**
* OSBS-8365

Signed-off-by: Robert Cerven <rcerven@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by twaugh** [comment]
Would be nicer if the koji_import plugin found it from the fetch_sources plugin result.

**Comment by rcerven** [comment]
> Would be nicer if the koji_import plugin found it from the fetch_sources plugin result.

changed

File: atomic_reactor/plugins/pre_fetch_sources.py [hunk 1]
```diff
@@ -217,4 +218,5 @@ def get_signing_intent(self):
                 self.signing_intent = odcs_config.default_signing_intent
 
         signing_intent = odcs_config.get_signing_intent_by_name(self.signing_intent)
+        self.workflow.koji_source_signing_intent = self.signing_intent
```
Comment (twaugh) [inline atomic_reactor/plugins/pre_fetch_sources.py]: No longer needed?

Comment (rcerven) [reply atomic_reactor/plugins/pre_fetch_sources.py]: yeah already removed :)

**Comment by twaugh** [comment]
@MartinBasti will you be added the documentation to koji.md for this?

**Comment by rcerven** [comment]
rn updated: add signing intent to koji build metadata for source containers

**Comment by MartinBasti** [comment]
> @MartinBasti will you be added the documentation to koji.md for this?

Done: #1332

**Pull request merged**